test/e2e/scheduler_predicates.go

```diff
@@ -900,16 +886,37 @@ func getRequestedCPU(pod v1.Pod) int64 {
 	return result
 }
 
-func waitForScheduler() {
-	// Wait a bit to allow scheduler to do its thing
-	// TODO: this is brittle; there's no guarantee the scheduler will have run in 10 seconds.
-	framework.Logf("Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.")
-	time.Sleep(10 * time.Second)
+func waitForScheduler(f *framework.Framework, PodName string, expectSuccess bool) {
+	framework.Logf("Waiting for scheduler...")
+	action := func() error { // No action before waiting for Scheduler.
+		return nil
```

Comment:
In case it's not clear (on re-read). The racy ordering we'd want to avoid is: 1) create the pod 2) scheduler emits a binding (and event) 3) create the watch 4) event observer predicate times out